qiskit_ibm_runtime/ibm_backend.py

```diff
@@ -233,7 +233,7 @@ def reservations(
         """
         start_datetime = local_to_utc(start_datetime) if start_datetime else None
         end_datetime = local_to_utc(end_datetime) if end_datetime else None
-        raw_response = self._api_client.backend_reservations(
+        raw_response = self._api_client.backend_reservations(  # type: ignore
```

Comment:
I don't understand why the constructor accepts two different clients (`Union[AccountClient, RuntimeClient]`) and this line invokes a function that only one of the clients implements. There might be implicit underlying assumptions but can those be made clearer?  E.g. by passing in `account_client` / `runtime_client` instead of `api_client` in the constructor?

Reply:
Thanks, I've split the parameters as suggested to make it more clear.